### PR TITLE
Correctly consider aiProcess_FlipWindingOrder and aiProcess_MakeLeftHanded when generating normals

### PR DESCRIPTION
**Pull request by lsnoel**
Fixes #3838

When assimp generates normals with aiProcess_GenNormals or aiProcess_GenSmoothNormals it does not take account of aiProcess_MakeLeftHanded, which flips the winding order and requires normals to be flipped.

Previous pull request #3838 fixed the same issue for aiProcess_FlipWindingOrder, but that broke normals for those using both flags (left handed and flipped winding order, typical for DirectX). When both flags are set, no action needs to be taken since the two normal flips cancel out. 

Corrected code flips normals if one or the other flag is set, but not both.

### CODE REVIEW

**Comment by Garux** [comment]
Looks legit 👍🏾

File: code/PostProcessing/GenFaceNormalsProcess.cpp [hunk 1]
```diff
@@ -131,8 +132,8 @@ bool GenFaceNormalsProcess::GenMeshFaceNormals(aiMesh *pMesh) {
         const aiVector3D *pV1 = &pMesh->mVertices[face.mIndices[0]];
         const aiVector3D *pV2 = &pMesh->mVertices[face.mIndices[1]];
         const aiVector3D *pV3 = &pMesh->mVertices[face.mIndices[face.mNumIndices - 1]];
-        if (flippedWindingOrder_)
-            std::swap( pV2, pV3 );
+        if (flippedWindingOrder_ != leftHanded_) // Boolean XOR
```
Comment (turol) [inline code/PostProcessing/GenFaceNormalsProcess.cpp:135]: Add comment explaining _why_.

File: code/PostProcessing/GenVertexNormalsProcess.cpp [hunk 2]
```diff
@@ -141,8 +142,8 @@ bool GenVertexNormalsProcess::GenMeshVertexNormals(aiMesh *pMesh, unsigned int m
         const aiVector3D *pV1 = &pMesh->mVertices[face.mIndices[0]];
         const aiVector3D *pV2 = &pMesh->mVertices[face.mIndices[1]];
         const aiVector3D *pV3 = &pMesh->mVertices[face.mIndices[face.mNumIndices - 1]];
-        if (flippedWindingOrder_)
-            std::swap( pV2, pV3 );
+        if (flippedWindingOrder_ != leftHanded_) // Boolean XOR
```
Comment (turol) [inline code/PostProcessing/GenVertexNormalsProcess.cpp:145]: Same

Comment (lsnoel) [reply code/PostProcessing/GenVertexNormalsProcess.cpp]: Done

**Review by kimkulling — APPROVED** [review]
Right, looks fine!

**Comment by kimkulling** [comment]
Merged, thanks a lot for your contribution.

**Pull request merged**